### ISSUE TITLE
Question about size conversion

### ISSUE THREAD

**Issue by Jixuying** [dianixn/Residual_CNN#1]
Thank you for your sharing your codes.

I am also studying the article"Deep Residual Learning Meets OFDM Channel Estimation" recently, and I noticed that deconvolution is used in this article to complete the final size conversion, and your code uses "resize2dLayer", that is, using interpolation to complete the size conversion.

I think the method of using deconvolution seems inflexible. If the number of pilots is different, the parameters of the deconvolution layer need to be reset and trained. I want to know why you didn't use the deconvolution layer in your code? Or have you compared it with the deconvolution method?

**Comment by dianixn** [comment]
Hi, conventionally deconvolution layer has parameters to learn, and that is the place neural network can work on. I am not sure the reason why they use deconvolution instead of interpolation because if you run a test you can see that from the result, the gain in the MSE plot is roughly same as the result from the paper. I had a check on the result from the channelnet paper and I think the MSE results are not matched, say starting from -2 level in the paper called Deep Residual Learning Meets OFDM Channel Estimation, but I think we all should a agreement on how LS implemented? If you know the reason for the mismatch hint me a bit.

**Comment by dianixn** [comment]
The last conv is used to deal with checkerboard artifacts potentially caused by deconvolution.

**Comment by Jixuying** [comment]
Hello, thank you very much for your reply!
Under the framework of PyTorch, I simply reproduced the results in the paper Deep ResidualLearning Meets OFDM Channel Estimation, and the network structure was set according to the description in the paper.
EPA model was selected as the channel model, and MSE loss function was adopted. Bilinear interpolation was used for size conversion as you did.
The MSE performance of both the training set and the verification set in the training process is better than the results in the paper (I haven't sorted out the data results of the test set).
Note that my performance abscissa is SIR, which means I'm adding the same frequency interference generated in the same way to my data set.

For the implementation of LS, I directly divide the received signal by the known transmitted signal at the pilot frequency, and then use linear interpolation to get the estimated channel value at other data locations.

**Comment by dianixn** [comment]
I mean everyone should have a common knowledge on LS so we can use LS as the reference. Can you achieve a similar dB gain in MSE plot compared with the diagram shown in that paper (MSE level from -2 for 0dB)? Sometimes MSE performance does not make much sense but gain make more sense. From my side, pytorch (also MSE) gives similar performance compared with MATLAB but could be that I was using commu.rayleigh not EPA.

Can I ask how you generate the same frequency interference, by sending a signal at the same frequency band through the channel and sum the user signal and interference? for the low SINR range, residual can still perform?

**Comment by dianixn** [comment]
Residual_transposed.m deploys the transpose convolution instead of resize.

**Comment by dianixn** [comment]
老哥，导频你咋放上去的，72*14里放48个，2个pilot symbols？然后一个里面放24个？
